### PR TITLE
Improvement on error reporting of 'for await' loops

### PR DESCRIPTION
**Pull request by adams85**
A minor fix related to async iterations (see https://github.com/sebastienros/esprima-dotnet/issues/183#issuecomment-901976754)

### CODE REVIEW

**Comment by adams85** [comment]
Ooops, looks like this needs some more work:

https://github.com/sebastienros/esprima-dotnet/pull/189/commits/af79159487a33cd23a97eb7e49134fde324eb49d#diff-92a67081d35facbe8921de0638d8eccd7f7204c8b99a656abc3af2ea750857c1R3045-R3055

Please don't merge this yet.

File: src/Esprima/JavascriptParser.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public Context()
             public bool AllowIn;
             public bool AllowStrictDirective;
             public bool AllowYield;
-            public bool Await;
```
Comment (lahma) [inline src/Esprima/JavascriptParser.cs:44]: even thought might not seem a good name, I think we should keep the naming that JS project has to make diffing and syncing easier

Comment (adams85) [reply src/Esprima/JavascriptParser.cs:44]: No problem, I'll revert it.

**Comment by adams85** [comment]
@lahma Ok, I went over it again, I think it's ready now.

**Comment by lahma** [comment]
I think we should wait for the Esprima change first, it also renames the `context.await` to `context.isAsync` which is more logical and will also reflect here (unfortunately clashing). I'd like to also see how this changes reported errors against JS test suite after those changes (ideally this would show improvements 😉 )

**Comment by sebastienros** [comment]
Fixed the conflict from the GH UI, hopefully correctly.

**Review by lahma — APPROVED** [review]
I think it's all good, might need some styling adjustment later on but test cases are covering now 👍🏻

**Pull request merged**